Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -258,6 +258,26 @@ open class ChatThreadVC: _ViewController,
         messageComposerVC.dismissSuggestions()
     }
 
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        decorationViewForMessage message: ChatMessage,
+        decorationType: ChatMessageDecorationType,
+        at indexPath: IndexPath
+    ) -> ChatMessageDecorationView? {
+        switch decorationType {
+        case .header where vc.shouldShowDateSeparator(forMessage: message, at: indexPath):
+            let dateSeparatorView = components.messageListDateSeparatorView.init()
+            dateSeparatorView.content = vc.dateSeparatorFormatter.format(message.createdAt)
+            return dateSeparatorView
+        case .footer where components.threadRepliesCounterEnabled && message == messages.last:
```

Comment:
Yeah, I think this verifies what I said in the previous PR. IMO having footer and header in separate delegate methods makes more sense